src/main/java/com/marklogic/kafka/connect/sink/WriteBatcherSinkTask.java

```diff
@@ -215,6 +269,26 @@ public void processFailure(WriteBatch batch, Throwable throwable) {
                             values.get("kafka-offset"), values.get("kafka-timestamp"), values.get("kafka-topic"));
                     }
                 }
+                if (writeHandle instanceof SinkRecordMetadataHandle) {
+                    SinkRecord sinkRecord = ((SinkRecordMetadataHandle) writeHandle).getMySinkRecord();
+                    if (sinkRecord instanceof InternalSinkRecord) {
```

Comment:
This looks identical to the block for conversion errors - should make this into a new private method to avoid the duplication.